lib/sidekiq/statistic/middleware.rb

```diff
@@ -30,25 +30,17 @@ def save_entry_for_worker(worker_status)
         status = worker_status.dup
         realtime_hash = "#{REDIS_HASH}:realtime:#{Time.now.sec}"
         worker_key = "#{Time.now.utc.to_date}:#{status.delete :class}"
-        time_keys = ["#{worker_key}:min_time", "#{worker_key}:max_time", "#{worker_key}:average_time"]
 
         Sidekiq.redis do |redis|
-          min_time, max_time, average_time =
-            redis.hmget(REDIS_HASH, time_keys).map{ |v| (v || status[:time]).to_f }
-          min_time, max_time = [min_time, max_time, status[:time]].minmax
-
           statistics = [
             "#{worker_key}:last_job_status", status[:last_job_status],
-            "#{worker_key}:average_time", (average_time + status[:time]) / 2,
-            "#{worker_key}:last_time", status[:last_runtime],
-            "#{worker_key}:min_time", min_time,
-            "#{worker_key}:max_time", max_time
+            "#{worker_key}:last_time", status[:last_runtime]
           ]
 
           redis.hincrby REDIS_HASH, "#{worker_key}:#{status[:last_job_status]}", 1
-          redis.hincrbyfloat REDIS_HASH, "#{worker_key}:total_time", status[:time]
 
           redis.hmset REDIS_HASH, statistics
+          redis.lpush "#{worker_key}:timeslist", status[:time]
```

Comment:
Some workers might be called millions of times in a day. That's a lot of data to save. Saving all data from the last hour seems ok.

> On Aug 15, 2015, at 10:34, Anton Davydov <notifications@github.com> wrote:
> 
> In lib/sidekiq/statistic/middleware.rb:
> 
> >  
> >            redis.hmset REDIS_HASH, statistics
> > +          redis.lpush "#{worker_key}:timeslist", status[:time]
> @mperham I want to save all job time in redis lists (list called <yyyy-mm-dd>:<WorkerName>:timeslist). Yes, for each day I have separate list for each worker. Do you think it's bad idea?
> 
> —
> Reply to this email directly or view it on GitHub.
>